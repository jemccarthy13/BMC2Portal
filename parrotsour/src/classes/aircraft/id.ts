export enum IDMatrix {
  HOSTILE = "red",
  SUSPECT = "orange",
  ASSUME_FRIEND = "green",
  NEUTRAL = "purple",
  FRIEND = "blue",
}
