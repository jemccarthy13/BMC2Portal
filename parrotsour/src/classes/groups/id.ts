export enum IDMatrix {
    HOSTILE = "red",
    SUSPECT = "#FA8072",
    ASSUME_FRIEND = "light green",
    NEUTRAL = "purple",
    FRIEND = "blue"
}