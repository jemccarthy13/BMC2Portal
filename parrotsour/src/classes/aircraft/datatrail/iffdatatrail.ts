import { Point } from "../../point"
import { IDMatrix } from "../id"
import { DataTrail } from "./datatrail"

export class IFFDataTrail extends DataTrail {
  private iffData: Point[] = []

  // eslint-disable-next-line @typescript-eslint/no-unused-vars
  getCenterOfMass(_heading: number): Point {
    return this.iffData[this.iffData.length - 1]
  }

  draw(ctx: CanvasRenderingContext2D, heading: number, id: IDMatrix): void {
    if (id !== IDMatrix.SUSPECT && id !== IDMatrix.HOSTILE) {
      //console.log("draw iff")
    }
    // error but don't break the runtime
  }
}
