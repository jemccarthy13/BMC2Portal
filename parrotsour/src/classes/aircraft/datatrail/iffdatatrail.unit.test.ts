import { Point } from "../../point"
import { IDMatrix } from "../id"
import { IFFDataTrail } from "./iffdatatrail"

// eslint-disable-next-line @typescript-eslint/no-var-requires, @typescript-eslint/no-unused-vars
const canvasSerializer = require("jest-canvas-snapshot-serializer")

const canvas = document.createElement("canvas")
// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
const ctx = canvas.getContext("2d")!

describe("IFFDataTrail", () => {
  it("draws_correctly", () => {
    const datatrail = new IFFDataTrail(new Point(50, 50), 90)
    datatrail.draw(ctx, 90, IDMatrix.FRIEND)
    expect(canvas).toMatchSnapshot()
  })

  it("doesnt_draw_hos_sus", () => {
    const datatrail = new IFFDataTrail(new Point(50, 50), 90)
    datatrail.draw(ctx, 90, IDMatrix.HOSTILE)
    datatrail.draw(ctx, 90, IDMatrix.SUSPECT)
    expect(canvas).toMatchSnapshot()
  })
})
