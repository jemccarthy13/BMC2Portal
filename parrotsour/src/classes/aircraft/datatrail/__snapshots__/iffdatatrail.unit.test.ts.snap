// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IFFDataTrail doesnt_draw_hos_sus 1`] = `<canvas />`;

exports[`IFFDataTrail draws_correctly 1`] = `<canvas />`;
